### PR TITLE
hallmarks_of_cancer dataset

### PR DESCRIPTION
**Pull request by debajyotidatta**

### CODE REVIEW

**Review by jason-fries — CHANGES_REQUESTED** [review]
@debajyotidatta Can we update this to the most recent schema format? :)

**Comment by sunnnymskang** [comment]
@debajyotidatta pinging again to adapt the schema to reflect the most recent format: https://github.com/bigscience-workshop/biomedical/blob/master/schemas/text.py

**Review by sunnnymskang — CHANGES_REQUESTED** [review]
HI @debajyotidatta Can you change the formatting to agree with the latest template changes? (https://github.com/bigscience-workshop/biomedical/blob/master/templates/template.py) Specifically, there's an explicit "source" in addition to "bigbio" Though it's a cosmetic change, I think it'll be helpful that all examples have uniform formatting that aligns with the template.py

File: examples/hallmarks_of_cancer.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+class Hallmarks_Of_Cancer(datasets.GeneratorBasedBuilder):
+    """Hallmarks Of Cancer Dataset"""
+
+    VERSION = datasets.Version(_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name=_DATASETNAME,
```
Comment (galtay-tempus) [inline examples/hallmarks_of_cancer.py:62]:
```suggestion
            name="source",
```

File: examples/hallmarks_of_cancer.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = (
+        _DATASETNAME  # It's not mandatory to have a default configuration. Just use one if it make sense.
+    )
+
+    def _info(self):
+
+        if self.config.name == _DATASETNAME:
```
Comment (galtay-tempus) [inline examples/hallmarks_of_cancer.py:80]:
```suggestion
        if self.config.name == "source":
```

File: examples/hallmarks_of_cancer.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+            label_file_name = labels[0] + '/' + filenname
+            with open(label_file_name, encoding='utf-8') as f:
+                lines = f.readlines()
+                label_body = "".join([j.strip() for j in lines])
+                label_body = [i.strip() for i in label_body.split('<')]
+                label_body = sum([k.split('AND') for k in label_body if len(k)>1], [])
+                label_body = [i.split('--')[0].strip() for i in label_body]
+
+            if self.config.name == _DATASETNAME:
```
Comment (galtay-tempus) [inline examples/hallmarks_of_cancer.py:157]:
```suggestion
            if self.config.name == "source":        
```

File: examples/hallmarks_of_cancer.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+text can be found under "text" directory. The filenames are the corresponding PubMed IDs (PMID).
+"""
+
+_HOMEPAGE = "https://github.com/sb895/Hallmarks-of-Cancer"
+
+_LICENSE = "GNU General Public License v3.0"
+
+_URLs = {"hallmarks_of_cancer": "https://github.com/sb895/Hallmarks-of-Cancer/archive/refs/heads/master.zip"}
+
+_VERSION = "1.0.0"
```
Comment (sunnnymskang) [inline examples/hallmarks_of_cancer.py:35]:
```suggestion
_SOURCE_VERSION = "1.0.0"
```

File: examples/hallmarks_of_cancer.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    'NULL',
+    'Sustaining proliferative signaling',
+    'Tumor promoting inflammation'
+]
+
+
+class Hallmarks_Of_Cancer(datasets.GeneratorBasedBuilder):
+    """Hallmarks Of Cancer Dataset"""
+
+    VERSION = datasets.Version(_VERSION)
```
Comment (sunnnymskang) [inline examples/hallmarks_of_cancer.py:57]:
```suggestion
    VERSION = datasets.Version(_SOURCE_VERSION)
```

**Review by sunnnymskang — CHANGES_REQUESTED** [review]
With these changes, I think it's ready to be merged

**Comment by debajyotidatta** [comment]
I think this is good to merge. Are we waiting for something?

**Review by galtay — APPROVED** [review]
looks good to me

**Comment by galtay** [comment]
going to merge this and we can address any problems that come up when unit tests are applied in subsequent PRs

**Pull request merged**